source/common/http/conn_manager_impl.cc

```diff
@@ -173,6 +173,9 @@ StreamDecoder& ConnectionManagerImpl::newStream(StreamEncoder& response_encoder)
   new_stream->response_encoder_ = &response_encoder;
   new_stream->response_encoder_->getStream().addCallbacks(*new_stream);
   config_.filterFactory().createFilterChain(*new_stream);
+  if (underlying_connection_above_high_watermark_) {
```

Comment:
Is this actually needed? Won't the callbacks fire when the filter registers for callbacks later? I'm wondering if we can kill underlying_connection_above_high_watermark_ variable entirely?